### PR TITLE
Help message

### PR DESCRIPTION
**Pull request by jjaykim**
# Befor putting PR (put x in bracket if you ran it.);

- [x] Run `pylint $(git ls-files '*.py')` passes

# Overview

remove default help command and then create our custom help command

https://github.com/dhmoon91/discord/issues/5

desktop version
macbookpro

mobile version
iphone

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -43,6 +44,53 @@ async def on_member_join(member):
...
 
 
+# remove default help command to use our help command
+bot.remove_command("help")
+
+# Custom help command
+@bot.command(name="help", help="This is help command")
+async def help_command(ctx):
+    """Help command outputs description about all the commands"""
+    # Create to display help command in embedded format
```
Comment (dhmoon91) [inline bot.py]: This comment doesn't make any sense

File: bot.py [hunk 2]
```diff
@@ -43,6 +44,53 @@ async def on_member_join(member):
     await member.dm_channel.send(f"Hi {member.name}, welcome to 관전남 월드!")
 
 
+# remove default help command to use our help command
+bot.remove_command("help")
+
+# Custom help command
+@bot.command(name="help", help="This is help command")
```
Comment (dhmoon91) [inline bot.py]: the `help` string can be more descriptive instead of `This is help command`....
Of course we know `This is help command`...

**Comment by dhmoon91** [comment]
We should indicate in help message that we use data from `NA` server

**Comment by dhmoon91** [comment]
All help message needs to be updated. 

eg;
```
@관전남-jj last_match [summoner name]
Get last match history
```

We are not 'getting' last match history. 
We **show** last_match history stat of summoner

File: bot.py [hunk 3]
```diff
@@ -43,6 +44,53 @@ async def on_member_join(member):
...
+        value=f"<@!{bot.user.id}> **{get_rank.name} [summoner name]** \n {get_rank.help}",
+        inline=False,
+    )
+
+    help_embed.add_field(
+        name="** **",
+        value=f"<@!{bot.user.id}> **{get_last_match.name} [summoner name]** \n \
+         {get_last_match.help}",
+        inline=False,
+    )
```
Comment (dhmoon91) [inline bot.py]: For these commands, we should make array of commands -> loop through them instead of hardcoding it one by one. 

eg; list = [help, get_rank, get_last_match] <- loop through this array and add it to embed.

File: bot.py [hunk 4]
```diff
@@ -43,6 +44,53 @@ async def on_member_join(member):
...
+
+    help_embed.add_field(
+        name="** **",
+        value=f"<@!{bot.user.id}> **{get_rank.name} [summoner name]** \n {get_rank.help}",
+        inline=False,
+    )
+
+    help_embed.add_field(
+        name="** **",
+        value=f"<@!{bot.user.id}> **{get_last_match.name} [summoner name]** \n \
```
Comment (dhmoon91) [inline bot.py]: let's remove bracket and just go with `summoner_name`.
People might get confused and put bracket around it like `rank [d4022]`

File: bot.py [hunk 5]
```diff
@@ -43,6 +44,53 @@ async def on_member_join(member):
     await member.dm_channel.send(f"Hi {member.name}, welcome to 관전남 월드!")
 
 
+# remove default help command to use our help command
+bot.remove_command("help")
```
Comment (dhmoon91) [inline bot.py]: why is this needed? what happens if you don't remove it?

**Comment by jjaykim** [comment]
> All help message needs to be updated.
> 
> eg;
> 
> ```
> @관전남-jj last_match [summoner name]
> Get last match history
> ```
> 
> We are not 'getting' last match history.
> We **show** last_match history stat of summoner

```
@bot.command(name="rank", help="Get rank of summoner")
async def get_rank(ctx, name: str):

@bot.command(name="rank", help="Get rank of summoner")
async def get_rank(ctx, name: str):

help_embed.add_field(
        value=f"<@!{bot.user.id}> **{get_rank.name} [summoner name]** \n {get_rank.help}",
    )

help_embed.add_field(
        value=f"<@!{bot.user.id}> **{get_last_match.name} [summoner name]** \n \
         {get_last_match.help}",
    )

```
as you can see these codes, I just quoted the help attribute of each command. should I create a different description of them or change each command's help attribute?

**Comment by dhmoon91** [comment]
> > All help message needs to be updated.
> > eg;
> > ```
> > @관전남-jj last_match [summoner name]
> > Get last match history
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > We are not 'getting' last match history.
> > We **show** last_match history stat of summoner
> 
> ```
> @bot.command(name="rank", help="Get rank of summoner")
> async def get_rank(ctx, name: str):
> 
> @bot.command(name="rank", help="Get rank of summoner")
> async def get_rank(ctx, name: str):
> 
> help_embed.add_field(
>         value=f"<@!{bot.user.id}> **{get_rank.name} [summoner name]** \n {get_rank.help}",
>     )
> 
> help_embed.add_field(
>         value=f"<@!{bot.user.id}> **{get_last_match.name} [summoner name]** \n \
>          {get_last_match.help}",
>     )
> ```
> 
> as you can see these codes, I just quoted the help attribute of each command. should I create a different description of them or change each command's help attribute?

yeah update each command's help attribute

File: bot.py (hunk 5 above)
Comment (jjaykim) [reply bot.py]: if we do not remove the default help command, an error will occur. 
I wil attach the screenshot of this error.
Screen Shot 2021-07-02 at 1 34 56 AM
The error said `The command help is already an existing command or alias.`

File: bot.py (hunk 1 above)
Comment (jjaykim) [reply bot.py]: I will change the comment to `"""Illustrate how to use the bot and explains all the commands"""`.
How about this?

File: bot.py (hunk 4 above)
Comment (jjaykim) [reply bot.py]: I removed bracket []

File: bot.py (hunk 1 above)
Comment (dhmoon91) [reply bot.py]: im talking about `Create to display help command in embedded format`

Comment (jjaykim) [reply bot.py]: I think this comment doesn't necessary. I will remove it.

File: bot.py (hunk 3 above)
Comment (jjaykim) [reply bot.py]: I changed it

**Review by dhmoon91 — APPROVED** [review]
